arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -834,14 +854,23 @@ void IResearchViewExecutor<ordered, materializeType>::fillBuffer(IResearchViewEx
     }
 
     if constexpr (materializeType == MaterializeType::LateMaterializedWithVars) {
-      TRI_ASSERT((!this->_infos.getOutNonMaterializedViewRegs().empty()));
-      if (ADB_LIKELY(!this->_infos.getOutNonMaterializedViewRegs().empty())) {
-        TRI_ASSERT(_sortReader);
-        irs::bytes_ref sortValue{irs::bytes_ref::NIL}; // sort column value
-        auto ok = _sortReader(_doc->value, sortValue);
+      auto const& columnsFieldsRegs = this->_infos.getOutNonMaterializedViewRegs();
+      TRI_ASSERT(!columnsFieldsRegs.empty());
+      std::unordered_map<int, irs::bytes_ref> storedValue;
+      irs::columnstore_reader::values_reader_f reader;
+      for (auto const& columnFieldsRegs : columnsFieldsRegs) {
+        if (columnFieldsRegs.first >= 0) { // not IResearchViewNode::SortColumnNumber
+          reader = _storedValueReaders[columnFieldsRegs.first];
```

Comment:
given `_storedValueReaders` is vector, can store pointer to `_sortReader` to avoid checking `columnFieldsRegs.first >= 0`?